nx/lib/nx.ex

```diff
@@ -5637,7 +5637,7 @@ defmodule Nx do
       {_, 0} -> multiply(t1, t2)
       {n, 1} -> dot(t1, [n - 1], t2, [0])
       {1, m} -> dot(t1, [0], t2, [m - 2])
-      {n, m} when n >= 2 and m >= 2 -> dot(t1, [n - 1], t2, [m - 2])
+      {n, m} when n >= 2 and m >= 2 -> dot(t1, [n - 1], nil, t2, [m - 2], nil)
```

Comment:
Please use an empty list instead of nil as that will give us more space for maneuver in the future.